File: src/test/java/com/edwardjones/cksn_ui_tests/tests/RemoveProductFromCartTest.java
import org.junit.Test;
import static org.junit.Assert.*;

public class RemoveProductFromCartTest {

    @Test
    public void testRemoveProduct() {
        ShoppingCart cart = new ShoppingCart();
        cart.addProduct("Product B");
        cart.removeProduct("Product B");

        assertFalse(cart.contains("Product B"));
        assertEquals(0, cart.getItemCount());
        assertEquals(0.0, cart.getTotalPrice(), 0.01);
    }
}

File: tests/remove_product_from_cart_test.py
import unittest

class TestShoppingCart(unittest.TestCase):

    def setUp(self):
        self.cart = ShoppingCart()
        self.cart.add_product("Product B")

    def test_remove_product(self):
        self.cart.remove_product("Product B")
        self.assertNotIn("Product B", self.cart.products)
        self.assertEqual(self.cart.item_count(), 0)
        self.assertEqual(self.cart.total_price(), 0)

if __name__ == '__main__':
    unittest.main()